src/databases/Cube/CubeReader.C

```diff
@@ -1,616 +1,758 @@
-      y_size = atoi( tokens[0].c_str() );
-      dy[0]  = atof( tokens[1].c_str() );
-      dy[1]  = atof( tokens[2].c_str() );
-      dy[2]  = atof( tokens[3].c_str() );
-  }else{
-     errorString = "Cube reader: Error in line 5: Too few entries to determine y grid (expected 4).";
-     return;
-  }
-  if (y_size<0){
-    y_size = -1 * y_size;
-    SetGridUnitsToBohr();
-    debug4 << "Setting grid units to bohr" << endl;
-  }
-  if ((dy[0]!=0)||(dy[2]!=0)) 
-    isSheared = true;
-
-  //6. line: Read the z grid
-  getline( file , header);
-  tokens.clear();
-  iss.clear();
-  iss.str(header);
-  copy(istream_iterator<string>(iss), istream_iterator<string>(), back_inserter<vector<string> >(tokens));
-  if( tokens.size() >= 4 ){
-      z_size = atoi( tokens[0].c_str() );
-      dz[0]  = atof( tokens[1].c_str() );
-      dz[1]  = atof( tokens[2].c_str() );
-      dz[2]  = atof( tokens[3].c_str() );
-  }else{
-     errorString = "Cube reader: Error in line 6: Too few entries to determine x grid (expected 4).";
-     return;
-  }
-  if (z_size<0){
-    z_size = -1 * z_size;
-    SetGridUnitsToBohr();
-    debug4 << "Setting grid units to bohr" << endl;
-  }
-  if ((dz[0]!=0)||(dz[1]!=0)) 
-    isSheared = true;
-  
-  debug4 << "Grid size is " << x_size << "x"
-      << y_size << "x"
-      << z_size << endl;
-  
-  debug4 << "Step size is"
-      << "(" << dx[0] << "," << dx[1] << "," << dx[2] << ") x "
-      << "(" << dy[0] << "," << dy[1] << "," << dy[2] << ") x "
-      << "(" << dz[0] << "," << dz[1] << "," << dz[2] << ")"
-      << endl;
-  
-  //Line 7 to 7+natoms : Read for each atom its type, isotope and x ,y , z location
-  atom_types.clear();
-  atom_locations.clear();
-
-  for (int i=0; i<natoms; i++) {
-    //float isotope;
-    
-    getline( file , header);
+// ****************************************************************************
+//  Method: CubeReader::readMetaData
+//
+//  Notes:
+//    Caller must open and close the file.
+//
+//  Purpose:
+//    Reads metadata from the file.
+//
+//  Modifications:
+//    Kathleen Biagas, Mon Jan 8, 2024
+//    Added logic to possibly read multiple lines of orbital information and
+//    to keep track of the offset (line number) for easier readback later.
+//
+// ****************************************************************************
+
+void
+CubeReader::readMetaData(bool keepFileOpen)
+{
+    //Open the file or if the file is already open, jump to the beginning of the file
+    if( !file.is_open() )
+    {
+        file.open( (char*) filename.c_str() );
+    }
+    else if( file.tellg() >0 )
+    {
+        file.seekg( 0 , ios::beg );
+    }
+    errorString = "";
+
+    // 1. line
+    // This is here used as an additional (non-standard) check for orbitals.
+    // Another check for the standard-compliant definition of orbitals is done
+    // at the end of this function
+    string header;
+    getline(file, header);
+    if (header.find("orbital")!=string::npos)
```

Comment:
I know it predates your changes here but it seems worthwhile for this to be case insensitive.

Reply:
FYI...I have taken to the practice of selecting `Approve` but with potentially useful feedback you can feel free to act on or ignore. If that is too confusing, please let me know. Otherwise, I am fine with whatever you finally choose to do.

Reply:
Actually, it is a bit confusing to be 'Approved' with suggested changes. If I make the suggested changes, I would need you to review again.

IMHO, it would be better to make the suggestions, even with the caveat that it could be left up to the author about following them. Then discussion could take place on those items, or committed, or new tickets created for follow up on the suggestions. Then the conversations could be closed properly.

Reply:
Any changes I make to code after an Approval will require re-review.

Reply:
Ok, so reading between the lines, you'd prefer I either `Approve` with no suggested changes or `Request changes` and not what I have been doing?

Reply:
Doesn't it seem a bit cleaner that way? If suggestions are provided but not required, and approval already given. I could commit the PR without discussing the suggestions and why I chose not to implement them. The conversations never get resolved in the PR. Perhaps the suggestions never get implemented or looked at again even if they are good and should be done. If, through the discussion of the suggestions, it is decided they aren't within the scope of the PR, they could be entered as tickets and then they would be tracked better.